core/Test/Tasty/Core.hs

```diff
@@ -303,16 +308,41 @@ after deptype s =
     Nothing -> error $ "Could not parse pattern " ++ show s
     Just e -> after_ deptype e
 
+-- | Run given 'TestTree's one after another.
+--
+-- @since 1.5
+afterTree
+  :: DependencyType -- ^ whether to run the tests even if some of the dependencies fail
+  -> TestTree -- ^ dependencies
+  -> TestTree -- ^ tests to run after all dependencies have finished
+  -> TestTree
+afterTree = AfterTree
+
+-- | Run given 'TestTree's one after another.
+--
+-- @since 1.5
+sequentialTestGroup
+  :: TestName -- ^ Test group name
+  -> DependencyType -- ^ whether to run the tests even if some of the dependencies fail
+  -> [TestTree] -- ^ tests to run sequentially
+  -> TestTree
+sequentialTestGroup groupName depType =
+  testGroup groupName . maybeToList . go
+ where
+  go [] = Nothing
+  go (t:ts) = Just (foldl' (afterTree depType) t ts)
+
 -- | An algebra for folding a `TestTree`.
 --
 -- Instead of constructing fresh records, build upon `trivialFold`
 -- instead. This way your code won't break when new nodes/fields are
 -- indroduced.
 data TreeFold b = TreeFold
-  { foldSingle :: forall t . IsTest t => OptionSet -> TestName -> t -> b
+  { foldSingle :: forall t . IsTest t => OptionSet -> ExactPath -> TestName -> t -> b
   , foldGroup :: OptionSet -> TestName -> b -> b
   , foldResource :: forall a . OptionSet -> ResourceSpec a -> (IO a -> b) -> b
   , foldAfter :: OptionSet -> DependencyType -> Expr -> b -> b
+  , foldAfterTree :: OptionSet -> DependencyType -> ExactPath -> b -> b
```

Comment:
From the name and the type of `AfterTree`, I would have expected the type of `foldAfterTree` to be `OptionSet -> DependencyType -> ExactPath -> b -> b -> b`. Does this change mean that the definition of `foldTestTree` is not really a catamorphism anymore? (tbh, I'm not aware if it even ever was one!)

Reply:
I ended up only using one, but I think it could take both. As for the catamorphism; I've tried to read up on it but I can't make heads or tails of it.

Reply:
> Does this change mean that the definition of foldTestTree is not really a catamorphism anymore? (tbh, I'm not aware if it even ever was one!)

`foldTestTree` is not a catamorphism even now: there is no direct correspondence between constructors of `TestTree` and fields of `TreeFold`.